corporate/lib/stripe.py

```diff
@@ -315,75 +315,71 @@ def stripe_get_customer(stripe_customer_id: str) -> stripe.Customer:
     )
 
 
-@catch_stripe_errors
-def do_create_stripe_customer(user: UserProfile, payment_method: Optional[str] = None) -> Customer:
-    realm = user.realm
-    # We could do a better job of handling race conditions here, but if two
-    # people from a realm try to upgrade at exactly the same time, the main
-    # bad thing that will happen is that we will create an extra stripe
-    # customer that we can delete or ignore.
-    stripe_customer = stripe.Customer.create(
-        description=f"{realm.string_id} ({realm.name})",
-        email=user.delivery_email,
-        metadata={"realm_id": realm.id, "realm_str": realm.string_id},
-        payment_method=payment_method,
-    )
-    stripe.Customer.modify(
-        stripe_customer.id, invoice_settings={"default_payment_method": payment_method}
-    )
-    event_time = timestamp_to_datetime(stripe_customer.created)
-    with transaction.atomic():
+class BillableUser:
+    def __init__(self, user: UserProfile) -> None:
+        self.user = user
+
+    def get_customer(self) -> Optional[Customer]:
+        return get_customer_by_realm(self.user.realm)
+
+    def create_audit_log(self, event_type: int, event_time: datetime) -> None:
```

Comment:
Maybe `write_to_audit_log` would be a clearer name? I realize the thing you do in the database is a `.objects.create()` call, but I think semantically that'd be a bit more ergonomic.

Reply:
Updated as suggested above in revised changes.